src/opera/util/input_validation.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+"""
+=================
+input_validation.py
+=================
+
+Common code used by some PGEs for input validation.
+
+"""
+
+from os.path import exists, splitext
+
+from opera.util.error_codes import ErrorCode
+
+
+def _check_input(input_object, logger, name, valid_extensions):
+    """
+    Called by _validate_inputs() to check individual files.
+    The input object is checked for existence and that it ends with
+    a valid file extension.
+
+    Parameters
+    ----------
+    input_object : str
+        Relative path the object to be validated
+    logger: pge logger
+        pge.logger passed by PGE
+    name: str
+        pge name
+    valid_extensions : iterable
+        Expected file extensions of the file being validated.
+
+    """
+    if not exists(input_object):
+        error_msg = f"Could not locate specified input {input_object}."
+        logger.critical(name, ErrorCode.INPUT_NOT_FOUND, error_msg)
+
+    ext = splitext(input_object)[-1]
+
+    if ext not in valid_extensions:
+        error_msg = f"Input file {input_object} does not have an expected file extension."
+        logger.critical(name, ErrorCode.INVALID_INPUT, error_msg)
+
+
+def validate_inputs(runconfig, logger, name):
```

Comment:
Can we give this a slightly more specific name? Since the function performs the input validation for the SLC-S1 based PGE's, maybe something like `validate_slc_s1_inputs`?